admittance_controller/include/admittance_controller/admittance_rule.hpp

```diff
@@ -0,0 +1,215 @@
+   *
+   * \param[in] current_joint_state
+   * \param[in] measured_wrench
+   * \param[in] reference_joint_state
+   * \param[in] period
+   * \param[out] desired_joint_state
+   */
+  controller_interface::return_type update(
+    const trajectory_msgs::msg::JointTrajectoryPoint & current_joint_state,
+    const geometry_msgs::msg::Wrench & measured_wrench,
+    const trajectory_msgs::msg::JointTrajectoryPoint & reference_joint_state,
+    const rclcpp::Duration & period,
+    trajectory_msgs::msg::JointTrajectoryPoint & desired_joint_states);
+
+  /**
+   * Set fields of `state_message` from current admittance controller state.
+   *
+   * \param[out] state_message
+   */
+  const control_msgs::msg::AdmittanceControllerState & get_controller_state();
+
+public:
+  // admittance config parameters
+  std::shared_ptr<admittance_controller::ParamListener> parameter_handler_;
+  admittance_controller::Params parameters_;
+
+protected:
+  /**
+   * Calculates the admittance rule from given the robot's current joint angles. The admittance controller state input
+   * is updated with the new calculated values. A boolean value is returned indicating if any of the kinematics plugin
+   * calls failed.
+   * \param[in] admittance_state
+   * \param[in] dt
+   * \param[out] success
+   */
+  bool calculate_admittance_rule(AdmittanceState & admittance_state, double dt);
+
+  /**
+   * Updates internal estimate of wrench in world frame `wrench_world_` given the new measurement `measured_wrench`,
+   * the sensor to base frame rotation `sensor_world_rot`, and the center of gravity frame to base frame rotation `cog_world_rot`.
+   * The `wrench_world_` estimate includes gravity compensation
+   * \param[in] measured_wrench
+   * \param[in] sensor_world_rot
+   * \param[in] cog_world_rot
+   */
+  void process_wrench_measurements(
+    const geometry_msgs::msg::Wrench & measured_wrench,
+    const Eigen::Matrix<double, 3, 3> & sensor_world_rot,
+    const Eigen::Matrix<double, 3, 3> & cog_world_rot);
+
+  template <typename T1, typename T2>
+  void vec_to_eigen(const std::vector<T1> & data, T2 & matrix);
+
+  // number of robot joint
+  size_t num_joints_;
+
+  // Kinematics interface plugin loader
+  std::shared_ptr<pluginlib::ClassLoader<kinematics_interface::KinematicsInterface>>
+    kinematics_loader_;
+  std::unique_ptr<kinematics_interface::KinematicsInterface> kinematics_;
+
+  // filtered wrench in world frame
+  Eigen::Matrix<double, 6, 1> wrench_world_;
+
+  // admittance controllers internal state
+  AdmittanceState admittance_state_{0};
+
+  // transforms needed for admittance update
+  AdmittanceTransforms admittance_transforms_;
+
+  // position of center of gravity in cog_frame
+  Eigen::Vector3d cog_pos_;
+
+  // force applied to sensor due to weight of end effector
+  Eigen::Vector3d end_effector_weight_;
+
+  // ROS
+  control_msgs::msg::AdmittanceControllerState state_message_;
+  std::shared_ptr<rclcpp::Clock> clock_;
```

Comment:
It was needed for TF, but that is no longer the case because all of the transforms are calculated from the kinematics interface now. I will delete it.